src/rdkafka_transport.c

```diff
@@ -1403,6 +1403,15 @@ static void rd_kafka_transport_io_event (rd_kafka_transport_t *rktrans,
 
 	case RD_KAFKA_BROKER_STATE_AUTH:
 		/* SASL handshake */
+
+        if (events & POLLHUP) {
+            rd_kafka_broker_fail(rkb, LOG_ERR,
```

Comment:
set `errno = EINVAL;` prior to calling broker_fail(). It logs the errno in debug messages and we want to have it set to a known value.

Reply:
Done as requested.